h2o-hadoop-2/assemblyjar.gradle

```diff
@@ -15,60 +15,60 @@ if (hasCustomHdfsDep) {
 }
 
 dependencies {
-  compile("org.apache.hadoop:hadoop-client:$hadoopMavenArtifactVersion") {
+  api("org.apache.hadoop:hadoop-client:$hadoopMavenArtifactVersion") {
     force = true
   }
-  compile("org.apache.hadoop:hadoop-common:$hadoopMavenArtifactVersion") {
+  api("org.apache.hadoop:hadoop-common:$hadoopMavenArtifactVersion") {
     force = true
   }
   if (hasCustomHdfsDep) {
-    compile("org.apache.hadoop:${hdfsDependency}:$hadoopMavenArtifactVersion") {
+    api("org.apache.hadoop:${hdfsDependency}:$hadoopMavenArtifactVersion") {
       force = true
     }
   }
-  compile(project(":h2o-mapreduce-generic")) {
+  api(project(":h2o-mapreduce-generic")) {
     transitive = false
   }
-  compile project(":h2o-security")
-  compile project(":h2o-ext-steam")
+  api project(":h2o-security")
+  api project(":h2o-ext-steam")
   // Libraries need for Google Cloud Storage strongly require this Guava version
-  compile('com.google.guava:guava:20.0') {force = true}
-  compile(project(':h2o-app')) {
+  api('com.google.guava:guava:20.0') {force = true}
+  api(project(':h2o-app')) {
     exclude module: "${defaultWebserverModule}"
   }
-  compile project(":h2o-web")
-  compile project(":h2o-avro-parser")
+  api project(":h2o-web")
+  api project(":h2o-avro-parser")
   // Include GCS persist layer
-  compile project(":h2o-persist-gcs")
+  api project(":h2o-persist-gcs")
   // Include S3 persist layer
-  compile project(":h2o-persist-s3")
+  api project(":h2o-persist-s3")
   // Include HDFS persist layer
-  compile(project(':h2o-persist-hdfs')) {
+  api(project(':h2o-persist-hdfs')) {
```

Comment:
IMHO, all dependecies 
`
compile(project(...)) {
    transitive = false
  }
  `
  could be changed  to:
  `
implementation(project(...)) {
    transitive = false
  }
  `